share/completions/gradle.fish

```diff
@@ -1,60 +1,281 @@
+    test \( -e ./build.gradle -a -f ./build.gradle \) -o \( -e ./build.gradle.kts -a -f ./build.gradle.kts \)
+end
+
+function __create_completion_cache_file
     # Set up cache directory
     if test -z $XDG_CACHE_HOME
         set XDG_CACHE_HOME $HOME/.cache/
     end
     mkdir -m 700 -p $XDG_CACHE_HOME/gradle-completions
 
-    set -l hashed_pwd (__fish_md5 -s $PWD)
-    set -l gradle_cache_file $XDG_CACHE_HOME/gradle-completions/$hashed_pwd
-    if not test -f $gradle_cache_file; or command test build.gradle -nt $gradle_cache_file
-        command gradle -q tasks 2>/dev/null | string match -r '^[[:alnum:]]+ - .*' | string replace ' - ' \t >$gradle_cache_file
+    string trim -- $XDG_CACHE_HOME/gradle-completions/(fish_md5 -s $argv[1] | string split ' = ')[2]
+end
+
+##############################
+# Configure Tasks Completion #
+##############################
+
+# Outside of a Project
+function __get_gradle_default_task_completion
+    if __contains_gradle_build
+        return
     end
-    cat $gradle_cache_file
+
+    string replace --all ' - ' \t -- "\
+        buildEnvironment - Displays all buildscript dependencies declared in root project.
+        components - Displays the components produced by root project.
+        dependencies - Displays all dependencies declared in root project.
+        dependencyInsight - Displays the insight into a specific dependency in root project.
+        dependentComponents - Displays the dependent components of components in root project.
+        help - Displays a help message.
+        init - Initializes a new Gradle build.
+        model - Displays the configuration model of root project.
+        projects - Displays the sub-projects of root project.
+        properties - Displays the properties of root project.
+        tasks - Displays the tasks runnable from root project.
+        wrapper - Generates Gradle wrapper files." \
+        | string trim
 end
 
-function __contains_gradle_build
-    test -f build.gradle
+complete --command 'gw' --command 'gradle' --command 'gradlew' \
+    --exclusive \
+    --arguments "(__get_gradle_default_task_completion)"
+
+# Inside of a Project
+function __get_gradle_task_completion
+    if not __contains_gradle_build
+        return
+    end
+
+    set -l gradle_cache_file (__create_completion_cache_file "{$PWD}-tasks")
+    if not command test -f $gradle_cache_file
+        command gradle -q tasks --all 2>/dev/null | string match --regex '^[a-z][A-z:]+.*' | string replace ' - ' \t >$gradle_cache_file
+    end
+
+    # return possible tasks
+    string trim -- <$gradle_cache_file
+end
+
+complete --command 'gw' --command 'gradle' --command 'gradlew' \
+    --condition "__contains_gradle_build" \
+    --exclusive \
+    --arguments "(__get_gradle_task_completion)"
+
+
+###############################
+# Configure Option Completion #
+###############################
+
+function __get_console_completion
+    string replace --all ' - ' \t -- "\
```

Comment:
Typically this will be expressed with something like

```fish
printf '%s\t%s\n' \
auto "Enable color and other rich output in the console output when the build process is attached to a console, or to generate plain text only when not attached to a console. This is the default when Gradle is attached to a terminal." \
plain "This option disables all color and other rich output in the console output. This is the default when Gradle is not attached to a terminal."
```

Also these string are *entirely too long* as a description. Please rewrite!